feat: DIA-2104: [FE] Sample task UI interactions

nick-skriabin opened the pull request:
Improvements to the Sample Dataset UI and Example data.

<img width="2445" alt="image" src="https://github.com/user-attachments/assets/37a8f0fb-4bc7-4a51-b67b-5242da658cab" />

<img width="2444" alt="image" src="https://github.com/user-attachments/assets/55974104-ccba-4b18-ad2d-fd49c8275635" />

This pull request includes various updates and refactoring to improve the `Button` component, enhance the `ImportPage`, and update the `SampleDatasetSelect` component. The most important changes include adding a new prop to the `Button` component, restructuring the `ImportPage` layout, and modifying the `SampleDatasetSelect` to include a group for select items.

Enhancements to `Button` component:

* Added `rawClassName` prop to the `Button` component to allow more flexible styling. [[1]](diffhunk://#diff-582827abc39080c4ae2aa299d75dcd835968549046e0a5f9a3279dc28827d427L8-R8) [[2]](diffhunk://#diff-582827abc39080c4ae2aa299d75dcd835968549046e0a5f9a3279dc28827d427L50-R59)

Improvements to `ImportPage`:

* Added `Spinner` to handle loading states and improved error handling for sample data. [[1]](diffhunk://#diff-69c27f83b5c2e14a1e4fb411596068407ea0fbb9cbb20adfefb11deb122789e4R18) [[2]](diffhunk://#diff-69c27f83b5c2e14a1e4fb411596068407ea0fbb9cbb20adfefb11deb122789e4R482-R517)
* Restructured layout to better organize elements and improve responsiveness. [[1]](diffhunk://#diff-69c27f83b5c2e14a1e4fb411596068407ea0fbb9cbb20adfefb11deb122789e4R392) [[2]](diffhunk://#diff-69c27f83b5c2e14a1e4fb411596068407ea0fbb9cbb20adfefb11deb122789e4L421-R424) [[3]](diffhunk://#diff-69c27f83b5c2e14a1e4fb411596068407ea0fbb9cbb20adfefb11deb122789e4L430-R437) [[4]](diffhunk://#diff-69c27f83b5c2e14a1e4fb411596068407ea0fbb9cbb20adfefb11deb122789e4L471-R458)

Updates to `SampleDatasetSelect`:

* Added `SelectGroup` to the `SelectContent` for better organization of select items. [[1]](diffhunk://#diff-a5b6a20921ce7c4eaa4b7db73462effa2e02d835b4098c40cb29649fee3754ecL1-R1) [[2]](diffhunk://#diff-a5b6a20921ce7c4eaa4b7db73462effa2e02d835b4098c40cb29649fee3754ecL33-R45)

Refactoring:

* Removed the `SimpleCard` wrapper from `CodeBlock` for a simpler implementation.

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

nick-skriabin committed:
UI improvements to the Sample dataset select

nick-skriabin committed:
Fine tuning

nick-skriabin committed:
Waiting state for loading data example

nick-skriabin committed:
Dropdown min width

yyassi-heartex commented on web/apps/labelstudio/src/components/Button/Button.jsx:
```diff
@@ -5,7 +5,7 @@ import { FormSubmissionContext } from "../Form/FormContext";
 
 export const Button = React.forwardRef(
-  ({ children, type, extra, className, size, waiting, icon, tag, look, ...rest }, ref) => {
+  ({ children, type, extra, className, rawClassName, size, waiting, icon, tag, look, ...rest }, ref) => {
```
funnily enough this is implied to be in `rest` as it was being passed down to block regardlress so i've used it as such before - but im glad you made it more explicit

yyassi-heartex commented on web/libs/ui/src/lib/code-block/code-block.tsx:
```diff
@@ -1,19 +1,10 @@
-import { SimpleCard } from "../simple-card";
```
if this component is no longer being used by any other component - lets make sure to remove it

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2104/sample-data-refinement

nick-skriabin committed:
Update sizes

MihajloHoma commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14352831656)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-dia-2104/sample-data-refinement'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14352831656

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14352831656

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2104/sample-data-refinement

nick-skriabin committed:
Fix button size

AndrejOros commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14358624975)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-dia-2104/sample-data-refinement'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14358624975

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14358624975
